### ISSUE TITLE
pub serve ignores changes in git dependency

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#1186]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [zoechi](https://github.com/zoechi)**
_Originally opened as dart-lang/sdk#21515_

----

I 
- checked out https://github.com/dart-lang/core-elements
- created a new branch
- changed `this.scrollTop = index \* this.height;` to `this.scrollTop = (index \* this.height).round();`
- checked in the modification
- changed the dependency in my app to a git dependency
&nbsp;&nbsp;core_elements:
&nbsp;&nbsp;&nbsp;&nbsp;git:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;url: ../../dart-lang/core-elements
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ref: core-list-dart-scrollToItem
- added a dependency_overrides
&nbsp;&nbsp;core_elements:
&nbsp;&nbsp;&nbsp;&nbsp;git:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;url: ../../dart-lang/core-elements
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ref: core-list-dart-scrollToItem
because other dependencies depend on hosted core-elements 

I'm not able to make `pub serve` execute the updated code

I also tried 
- kill pub serve after modifying the file
- using a path dependency
- execute `pub cache repair`
but that didn't change anything.

I somehow managed to get the updated version while trying out different variants of the dependency but it contained a typo.
After fixing the typo I wasn't able to make the app execute another version than the one containing the typo.

**Comment by DartBot** [comment]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [zoechi](https://github.com/zoechi)**

----

I purged the content of `~/.pub-cache` and `~/.pubserve`, I also deleted the branch where I made the change, checked out master and fetched the updates from the GitHub repo where the bug was fixed in the meantime, but it still executes the code with the typo. I also tried making it a path dependency without any success.

**Comment by DartBot** [comment]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [zoechi](https://github.com/zoechi)**

----

I just changed it back (from a path) to a git dependency again, and now the change takes effect.

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

Can you provide steps that reproduce this from scratch?

Also, what version of pub are you using?
______
*Added [NeedsInfo](../labels/NeedsInfo) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [zoechi](https://github.com/zoechi)**

----

Dart VM version: 1.8.0-edge.41523 (Wed Nov  5 08:13:39 2014) on "linux_x64"

I tried to provide the steps above. I don't know what more details I can provide.
I'm using WebStorm 9.0.1 but it shouldn't matter because it also uses `pub serve` but who knows. 

I had a related problem not too long ago with this repo
https://github.com/dart-lang/googleapis_auth/issues/5#issuecomment-57487927

I checked out the repo to test an added feature which was not yet included in the published package.
A path dependency to the checked out Git repo didn't work. When executing the app the new feature was not included. Changing the dependency to a git dependency worked.
I had expected that for a path dependency just the files of the referenced directory are used but this doesn't seem to be the case.

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

> I tried to provide the steps above. I don't know what more details I can provide.

If those are the entire steps for reproduction, you probably need to run "pub update" to get the newest git revision. Git revisions are pinned just like versions for hosted packages.

> I had expected that for a path dependency just the files of the referenced directory are used but this doesn't seem to be the case. 

Can you file a separate issue for this with reproduction steps?

**Comment by DartBot** [comment]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [zoechi](https://github.com/zoechi)**

----

I run `pub upgrade` every time after modifying a dependency in `pubspec.yaml`.
Sorry if the description is not detailed enough. My brain still starts hurting when I think about this problem. I have tried everything I could think of for about 4 hours.
I'll think about what additional information I can provide but I need some distance before I can cope with this again.

**Comment by DartBot** [comment]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [zoechi](https://github.com/zoechi)**

----

Possible dup of http://dartbug.com/21390

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

I don't think this is the same as issue dart-lang/pub#1176. It's possible it's the same as issue dart-lang/sdk#21810, though; can you still reproduce it with the latest bleeding edge?

**Comment by DartBot** [comment]
<a href="https://github.com/zoechi"><img src="https://avatars.githubusercontent.com/u/405837?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [zoechi](https://github.com/zoechi)**

----

Can't tell which of both is more similar, for both `.pub` seems to be involved.

I had this issue only once but I pulled out the last few remaining hair of mine while trying to make it work. 
Therefore I thought it would help to create an issue and see if others run into this as well. 

I think you can close this issue.

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

*Added [CannotReproduce](../labels/CannotReproduce) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

Issue dart-lang/sdk#21848 has been merged into this issue.
